public/app/plugins/panel/table/table-new/TableCellOptionEditor.tsx

```diff
@@ -51,7 +51,7 @@ export const TableCellOptionEditor = ({ value, onChange }: Props) => {
   // When options for a cell change we merge
   // any option changes with our options object
   const onCellOptionsChange = (options: TableCellOptions) => {
-    settingCache[value.type] = merge(value, options);
+    settingCache[value.type] = merge({}, value, options);
```

Comment:
this was kind of interesting. We found in the bug bash that the `componentDidUpdate` lifecycle method in Sparkline was failing to detect a diff for the change to the config when we swapped between modes of Line rendering for Sparkline cells. Using the `structureRev` as a key to clobber the whole RDG does work, but I figured there was some mutation of a source object happening that caused this, and sure enough, we `merge` directly onto `value` here, which makes that method unable to compare because the previous props were mutated.